run-qa-check.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+set -e
+
+function remove_db() {
+  # Remove the database created during test to ensure other
+  # tests are not hindered from another test creating
+  # the database.
+  rm ./tests/openwisp2/openwisp-controller.db
```

Comment:
Are you sure this is needed?

The django tests do not touch the main database.
When using MySQL and PostgreSQL, a new DB named `{db_name}_test` is created and then destroyed, while when using sqlite, the database is run in memory: https://docs.djangoproject.com/en/dev/ref/settings/#std:setting-TEST_NAME

Maybe you didn't know this.

Reply:
If a database exists before running this script, say because of manual testing before automated testing,created by command: `python manage.py migrate`, then `SAMPLE_APP=1 python manage.py makemigrations` would fail with error.

```
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/home/atb00ker/.local/share/virtualenvs/openwisp-controller-JE1Ysw5j/lib/python3.7/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
    utility.execute()
  File "/home/atb00ker/.local/share/virtualenvs/openwisp-controller-JE1Ysw5j/lib/python3.7/site-packages/django/core/management/__init__.py", line 395, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/atb00ker/.local/share/virtualenvs/openwisp-controller-JE1Ysw5j/lib/python3.7/site-packages/django/core/management/base.py", line 328, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/atb00ker/.local/share/virtualenvs/openwisp-controller-JE1Ysw5j/lib/python3.7/site-packages/django/core/management/base.py", line 369, in execute
    output = self.handle(*args, **options)
  File "/home/atb00ker/.local/share/virtualenvs/openwisp-controller-JE1Ysw5j/lib/python3.7/site-packages/django/core/management/base.py", line 83, in wrapped
    res = handle_func(*args, **kwargs)
  File "/home/atb00ker/.local/share/virtualenvs/openwisp-controller-JE1Ysw5j/lib/python3.7/site-packages/django/core/management/commands/makemigrations.py", line 101, in handle
    loader.check_consistent_history(connection)
  File "/home/atb00ker/.local/share/virtualenvs/openwisp-controller-JE1Ysw5j/lib/python3.7/site-packages/django/db/migrations/loader.py", line 299, in check_consistent_history
    connection.alias,
django.db.migrations.exceptions.InconsistentMigrationHistory: Migration connection.0001_initial is applied before its dependency sample_config.0001_initial on database 'default'.
```

to prevent that, I added this here, an alternative would be to change the location the of existing database when called with command:

```
SAMPLE_APP=1 openwisp-qa-check \
    --migration-path "./tests/openwisp2/sample_config/migrations/" \
    --migration-module sample_config
```

Reply:
@atb00ker:

```
Migration connection.0001_initial is applied before its dependency sample_config.0001_initial on database 'default'
```

Why `connection.0001_initial` is dependent on `sample_config.0001_initial`?
Should't then also connection be named `sample_connection.0001_initial`?

Reply:
It's not required to extend `connections` module in order to extend `config` module, hence for keeping minimum changes in the example, I didn't extend it, should I create another project with extends all the modules? :smile:

Reply:
If the goal is to ensure extensibility work, we should write the tests in a way which would spot any potential error or mistake, so extending every module sounds necessary to me if we want to be 100% sure